src/SchedulerSubsystem/Scheduler.java

```diff
@@ -62,6 +89,39 @@ else if (event instanceof DestinationEvent)
             throw new InvalidTypeException("Event type received cannot be handled by this subsystem.");
     }
 
+    /**
+     * Filters destination floors from destination events.
+     *
+     * @param events List of destination events.
+     * @return List of destination floors.
+     */
+    private List<Integer> toDestinationFloors(List<DestinationEvent> events) {
```

Comment:
This removes direction from the matching. I think if you matched with the whole event, you can include direction checking without having to filter destination events. Currently you do not check for direction at all in "isElevatorStopping" from my understanding.

Reply:
Direction should be derived from the passengers in the elevator, it might get removed from the event soon in the next rework.

Reply:
Direction checking wasn't mentioned in the state diagram. Can you update the state diagram so I can implement your suggestions properly?

Scheduler State Diagram: https://app.diagrams.net/#G14DeIsZoibYy-BnQ7c81sKBTpHJQuF3Pq#%7B%22pageId%22%3A%22T1rtvHDK5Ro4PfxUw1XM%22%7D

Reply:
We talked about this during the meeting, Arthur brought it up. I made an amendment it's in the picture you took. Implementation wise, there's not much work to do except find the elevator direction which I mentioned in the comments above. Arthur implemented a method to do that in Elevator which you can copy over until we think of a better way to share the methods.